lib/matplotlib/axes/_axes.py

```diff
@@ -1825,40 +1827,49 @@ def bar(self, left, height, width=0.8, bottom=None, **kwargs):
         edgecolor : scalar or array-like, optional
             the colors of the bar edges
 
-        linewidth : scalar or array-like, optional, default: None
+        linewidth : scalar or array-like, optional
             width of bar edge(s). If None, use default
             linewidth; If 0, don't draw edges.
+            default: None
 
-        xerr : scalar or array-like, optional, default: None
+        xerr : scalar or array-like, optional
             if not None, will be used to generate errorbar(s) on the bar chart
+            default: None
 
-        yerr : scalar or array-like, optional, default: None
+        yerr : scalar or array-like, optional
             if not None, will be used to generate errorbar(s) on the bar chart
+            default: None
 
-        ecolor : scalar or array-like, optional, default: None
+        ecolor : scalar or array-like, optional
             specifies the color of errorbar(s)
+            default: None
 
-        capsize : integer, optional, default: 3
+        capsize : integer, optional
            determines the length in points of the error bar caps
+           default: 3
 
-        error_kw :
+        error_kw : dict, optional
             dictionary of kwargs to be passed to errorbar method. *ecolor* and
             *capsize* may be specified here rather than as independent kwargs.
 
-        align : ['edge' | 'center'], optional, default: 'edge'
-            If `edge`, aligns bars by their left edges (for vertical bars) and
-            by their bottom edges (for horizontal bars). If `center`, interpret
+        align : {'edge',  'center'}, optional
+            If 'edge', aligns bars by their left edges (for vertical bars) and
+            by their bottom edges (for horizontal bars). If 'center', interpret
             the `left` argument as the coordinates of the centers of the bars.
+            To align on the align bars on the right edge pass a negative
+            `width`.
 
-        orientation : 'vertical' | 'horizontal', optional, default: 'vertical'
+        orientation : {'vertical',  'horizontal'}, optional
             The orientation of the bars.
```

Comment:
That is the convention (first is default) specified in the numpydoc style
guide

On Sun, Apr 5, 2015, 14:50 Eric Firing <notifications@github.com> wrote:

> In lib/matplotlib/axes/_axes.py
> <https://github.com/matplotlib/matplotlib/pull/4196#discussion_r27779543>:
>
> >
> > -        orientation : 'vertical' | 'horizontal', optional, default: 'vertical'
> > +        orientation : {'vertical',  'horizontal'}, optional
> >              The orientation of the bars.
>
> A clear indication of the default is critical; I wonder whether this can
> be handled by scrupulously adhering to a convention in which the first
> alternative is the default.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4196/files#r27779543>.
>